modules/serverless-kn-func-invoke-reference.adoc

```diff
@@ -0,0 +1,111 @@
+// Module included in the following assemblies:
+//
+// * serverless/reference/kn-func-ref.adoc
+
+:_content-type: REFERENCE
+[id="serverless-kn-func-invoke-reference_{context}"]
+= kn func invoke optional parameters
+
+You can specify optional parameters for the sent request by using the `kn func invoke` CLI command flags.
+
+* `-t`, `--target`: Specifies the target instance of the invoked function, for example, `local` or `remote` or `https://staging.example.com/`. The default target is `local`.
+* `-f`, `--format`: Specifies the format of the message, for example, `cloudevent` or `http`.
+* `--id`: A unique string identifier for the request.
+* `-n`, `--namespace`: The namespace on the cluster.
+* `--source`: Sender name for the request. This corresponds to the CloudEvent `source` attribute.
+* `--type`: Type of the request, for example, `boson.fn`. This corresponds to the CloudEvent `type` attribute.
+* `--data`: Content for the request. For CloudEvent requests, this is the CloudEvent `data` attribute.
+* `--file`: Path to a local file containing data to be sent.
+* `--content-type`: The MIME content type for the request.
+* `-p`, `--path`: Path to the project directory.
+* `-c`, `--confirm`: Prompt to interactively confirm all options.
+* `-v`, `--verbose`: Print verbose output.
+* `-h`, `--help`: Prints information of usage of `kn func invoke`.
+
+In particular, you might find it useful to specify the following parameters:
+
+Event target:: The target instance of the invoked function. Accepts the `local` value for a locally deployed function, the `remote` value for a remotely deployed function, or a URL for a function deployed at an arbitrary endpoint. If target is not specified, it defaults to `local`.
+Event message format:: The message format, such as `http` or `cloudevent`. Defaults to the format of the template that was used when creating the function.
+Event type:: The type of event being sent. You can find information about the `type` parameter that is set for events from a certain event producer in the documentation for that event producer. For example, the API server source may set the `type` parameter of produced events as `dev.knative.apiserver.resource.update`.
```

Comment:
You could minimise "You can find information about the type parameter that is set for events from a certain event producer in the documentation for that event producer" as follows:

> "You can find information about the type parameter that is set in the documentation for each event producer."

Reply:
You could replace "may" with "might". The ISG recommends only using "may" to indicate permission.